tests/helics/application_api/CMakeLists.txt

```diff
@@ -39,41 +38,30 @@ heat-transfer-tests.cpp
 ValueFederateExtendedTests.cpp
 )
 
-#  QUICK API TESTS
-
-add_executable(application-api-tests ${application_api_test_sources} ${application_api_test_headers})
+add_executable(application-api-tests ${application_api_test_sources} ${application_api_test_headers} ${key_test_files} ${extended_test_sources})
 target_link_libraries(application-api-tests helics-static helics_test_base)
 
-#  KEY TESTS
-add_executable(key-tests key-tests.cpp ${key_test_files} ${application_api_test_headers})
-target_link_libraries(key-tests helics-static helics_test_base)
-
-
-#  EXTENDED TESTS
-add_executable(extended-api-tests ${application_api_test_sources} ${extended_test_sources} ${application_api_test_headers})
-
-target_link_libraries(extended-api-tests helics-static helics_test_base)
-
-set_target_properties (application-api-tests extended-api-tests key-tests PROPERTIES FOLDER tests)
+set_target_properties (application-api-tests PROPERTIES FOLDER tests)
 
 target_compile_definitions(application-api-tests PRIVATE "-DHELICS_BROKER_LOCATION=\"${HELICS_BROKER_LOC}\"")
 target_compile_definitions(application-api-tests PRIVATE "-DHELICS_INSTALL_LOCATION=\"${CMAKE_INSTALL_PREFIX}\"")
-
 target_compile_definitions(application-api-tests PRIVATE "-DTEST_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}\"")
-
-target_compile_definitions(extended-api-tests PRIVATE "-DHELICS_BROKER_LOCATION=\"${HELICS_BROKER_LOC}\"")
-target_compile_definitions(extended-api-tests PRIVATE "-DHELICS_INSTALL_LOCATION=\"${CMAKE_INSTALL_PREFIX}\"")
-
-target_compile_definitions(extended-api-tests PRIVATE "-DTEST_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}\"")
 target_compile_definitions(application-api-tests PRIVATE "-DQUICK_TESTS_ONLY")
 
-target_compile_definitions(key-tests PRIVATE "-DKEY_TESTS")
```

Comment:
also the number of tests run under KEY_TESTS is different than before as well, since this definition is no longer applied

Reply:
I thought I’d marked all of the cases covered by KEY_TESTS with the “key” label. Could the change in the data test case have affected the number of tests run? Another question - are the key tests intended to be a subset of the quick/regular tests? If there are some that are key test only, that could add a lot of labels.

Reply:
It is driven by the data test cases,  basically a different test set applied to the data test cases.  Ignoring the possibilities of TCP_DISABLE and no ZMQ_ENABLED.  KEY_TESTS were a strict subset.